src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java

```diff
@@ -74,24 +96,128 @@ public BaseType getType() {
+        if (dataValue.has(UPLOAD_TIMESTAMP)) {
+            final JsonNode uploadTimeStamp = dataValue.get(UPLOAD_TIMESTAMP);
+            validationResults = validateUploadTimeStamp(dataFieldId,uploadTimeStamp,caseFieldDefinition);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
             }
         }
 
         return Collections.emptyList();
     }
 
+    private List<ValidationResult> validateDocumentBinaryURL(final String dataFieldId,
+                                                             final JsonNode documentBinaryUrl,
+                                                             Pattern urlPattern,
+                                                             String documentUrlValue,
+                                                             String urlPatternString) {
+        isNullOrEmptyResult = isNullOrEmpty(documentBinaryUrl,DOCUMENT_BINARY_URL,dataFieldId);
+        if (!isNullOrEmptyResult.isEmpty()) {
+            return isNullOrEmptyResult;
+        }
+
+        final String documentBinaryUrlValue = documentBinaryUrl.textValue();
+        final Matcher documentBinaryUrlMatcher = urlPattern.matcher(documentBinaryUrlValue);
+
+        if (!documentBinaryUrlMatcher.matches()) {
+            LOG.error("{} does not match Document Management domain or expected URL path {} {}",
+                documentUrlValue, urlPatternString, dataFieldId);
+            return Collections.singletonList(new ValidationResult(documentBinaryUrlValue
+                + " does not match Document Management domain or expected URL path", dataFieldId));
+        }
+        return Collections.emptyList();
+    }
+
+    private List<ValidationResult> validateCategoryId(final String dataFieldId,
+                                                      final JsonNode dataValue,
+                                                      final CaseFieldDefinition caseFieldDefinition) {
+        final JsonNode categoryId = dataValue.get(CATEGORY_ID);
+        isNullOrEmptyResult = (isNullOrEmpty(categoryId,CATEGORY_ID,dataFieldId));
+        if (!isNullOrEmptyResult.isEmpty()) {
+            return isNullOrEmptyResult;
+        }
+
+        if (!dataValue.has("parent_category_id")) {
+            LOG.warn("Object does not have parent_category_id key specified");
+        }
+
+        validationResults =
+            textValidator.validate(dataFieldId, categoryId, caseFieldDefinition);
+        if (!checkValidationResults(validationResults,CATEGORY_ID).isEmpty()) {
+            return validationResults;
+        }
+
+        final List<Category> categoryList =
+            caseDefinitionRepository.getCaseType(caseFieldDefinition.getCaseTypeId()).getCategories();
+        final boolean categoryCheck = categoryList.stream()
+            .anyMatch(category ->
+                category.getCategoryId().contains(categoryId.textValue()));
```

Comment:
For a slightly more meaningful variable name:
```suggestion
       boolean caseTypeContainsCategoryId = categoryList.stream()
            .anyMatch(category ->
                category.getCategoryId().contains(categoryId.textValue()));
```
Also might be worth setting `categoryId.textValue()` to a variable before this `stream`, since we'll be getting the same value repeatedly